### PR TITLE
add gql queries and hook up data

### PR DESCRIPTION
**Pull request by jaserud**
**Description of changes**
- moved tabled info header donor data summary to seperate file
- added a query to get data for table and table header
- modified some of the components as needed for data hook up
- updated types to match elastic search mapping

Related gateway PR: https://github.com/icgc-argo/platform-api/pull/180

**Type of Change**

- [ ] Bug
- [ ] Styling
- [x] New Feature

**Checklist before requesting review:**

- [ ] Matches design:

  - component sizes, spacing, and styles
  - font size, weight, colour
  - spelling has been double checked

- [ ] New uikit components have Storybook stories
- [ ] Feature is minimally responsive.
- [ ] Manual testing of UI feature.
- [ ] Selenium test is completed and passing.

### CODE REVIEW

File: components/pages/submission-system/program-dashboard/DonorDataSummary/index.tsx [hunk 1]
```diff
@@ -45,23 +71,27 @@ const emptyState = (
   </NoData>
 );
 
-const readyState = (
+const readyState = (donorSummaries: any) => (
```
Comment (hlminh2000) [inline components/pages/submission-system/program-dashboard/DonorDataSummary/index.tsx]: why `any` if we know what the data looks like?

File: components/pages/submission-system/program-dashboard/DonorDataSummary/types.tsx [hunk 2]
```diff
@@ -1,27 +1,51 @@
-export type DonorEntityRecord = {
-  releaseState: ReleasedFilesState;
+// Elasticsearch mapping of the donor summary document
```
Comment (hlminh2000) [inline components/pages/submission-system/program-dashboard/DonorDataSummary/types.tsx]: UI is unaware of Elasticsearch don't think we need this comment eh

File: components/pages/submission-system/program-dashboard/DonorDataSummary/index.tsx (hunk 1 above)
Comment (hlminh2000) [reply components/pages/submission-system/program-dashboard/DonorDataSummary/index.tsx]: also we can just turn this into a component eh

File: components/pages/submission-system/program-dashboard/DonorDataSummary/index.tsx [hunk 3]
```diff
@@ -45,23 +71,27 @@ const emptyState = (
...
+  const { data: { programDonorSummaryEntries = [] } = {} } = useProgramDonorsSummaryQuery(
+    programShortName,
+  );
   return (
     <DashboardCard>
       <Typography variant="default" component="span">
         Donor Data Summary
       </Typography>
-      {DASHBOARD_ENABLED ? readyState : emptyState}
+      {DASHBOARD_ENABLED ? readyState(programDonorSummaryEntries) : emptyState}
```
Comment (hlminh2000) [inline components/pages/submission-system/program-dashboard/DonorDataSummary/index.tsx]: Should we just remove the `DASHBOARD_ENABLED` and let the table render the empty state? it has that built-in... Also can feed it the loading state

File: components/pages/submission-system/program-dashboard/DonorDataSummary/types.tsx (hunk 2 above)
Comment (jaserud) [reply components/pages/submission-system/program-dashboard/DonorDataSummary/types.tsx]: Yea, your right this was really more for me as I was working. Will remove.

File: components/pages/submission-system/program-dashboard/DonorDataSummary/index.tsx (hunk 1 above)
Comment (jaserud) [reply components/pages/submission-system/program-dashboard/DonorDataSummary/index.tsx]: True, the summary table is already a component might as well use that. Also I'll turn the emptystate into a component as well.

File: components/pages/submission-system/program-dashboard/DonorDataSummary/index.tsx (hunk 3 above)
Comment (jaserud) [reply components/pages/submission-system/program-dashboard/DonorDataSummary/index.tsx]: If we remove the flag I guess we'll just get the empty data from gateway eh which should be ok. Guess we can remove it.

Comment (hlminh2000) [reply components/pages/submission-system/program-dashboard/DonorDataSummary/index.tsx]: yeah the empty data will be equivalent to the empty state logically :P

Comment (jaserud) [reply components/pages/submission-system/program-dashboard/DonorDataSummary/index.tsx]: Right :heavy_check_mark:. Also good call I forgot about the loading state :man_facepalming:, will add.

File: components/pages/submission-system/program-dashboard/DonorDataSummary/DonorSummaryTableStatArea.tsx [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+  partiallyReleasedDonorsCount: 0,
+};
+
+const DonorStatsArea = ({
+  programDonorSummaryStats = emptyProgramDonorSummaryStats,
+}: {
+  programDonorSummaryStats: ProgoramDonorReleasStats;
+}) => {
+  const total = Object.values(programDonorSummaryStats)
+    .filter(Number)
```
Comment (hlminh2000) [inline components/pages/submission-system/program-dashboard/DonorDataSummary/DonorSummaryTableStatArea.tsx]: you mean `.map`?

Comment (jaserud) [reply components/pages/submission-system/program-dashboard/DonorDataSummary/DonorSummaryTableStatArea.tsx]: I'm not trying to convert to number, I want to filter the stuff that is a Number. I had to do this cause the gql object has a _type_something

Comment (hlminh2000) [reply components/pages/submission-system/program-dashboard/DonorDataSummary/DonorSummaryTableStatArea.tsx]: I would just explicitly write out `fullyReleasedDonorsCount + noReleaseDonorsCount + partiallyReleasedDonorsCount`, it's more reflective of the actual logic. 
We're not trying to summarize a hash map, we actually care about this specific sum eh

Comment (hlminh2000) [reply components/pages/submission-system/program-dashboard/DonorDataSummary/DonorSummaryTableStatArea.tsx]: reading the reduce does not tell me what's being added. Also we can't safely add a new field to the query without affecting this calculation.

Comment (hlminh2000) [reply components/pages/submission-system/program-dashboard/DonorDataSummary/DonorSummaryTableStatArea.tsx]: alternatively, the `registeredDonorsCount` gives the total donor count and avoids the sum altogether :P

Comment (jaserud) [reply components/pages/submission-system/program-dashboard/DonorDataSummary/DonorSummaryTableStatArea.tsx]: lol I didn't know that gave us the total donor count. Will update :+1:

**Pull request merged**